GLTF/src/GLTFMaterial.cpp

```diff
@@ -879,19 +921,19 @@ GLTF::MaterialPBR* GLTF::MaterialCommon::getMaterialPBR(GLTF::Options* options)
 		}
 	}
 
-	if (values->emission) {
-		if (values->emission[3] < 1.0) {
-			hasTransparency = true;
-		}
-		material->emissiveFactor = values->emission;
-	}
 	if (values->emissionTexture) {
 		GLTF::MaterialPBR::Texture* texture = new GLTF::MaterialPBR::Texture();
 		texture->texCoord = values->emissionTexCoord;
 		texture->texture = values->emissionTexture;
 		material->emissiveTexture = texture;
-		material->emissiveFactor = new float[3]{ 1.0, 1.0, 1.0 };
+        values->emission = new float[3]{ 1.0, 1.0, 1.0 };
```

Comment:
`values->emission` is expected to have four elements. As this is, the check for `values->emission[3]` below will go past the bounds of the array.

What was the intent here?

Reply:
So my intent was that values always hold the `emission` value. Sometimes we assign the `value->emission` to `material->emissiveFactor` but in cases of `emission` being a texture then we create a float array on the fly, so when we cleanup we don't know who owns it. If `values` always owns it, then it is easy to clean up.

Reply:
Also, I can just make it 4 like the emission is when specified from collada.

Reply:
If `values->emission` wasn't equal to `emissiveFactor` then I delete it, otherwise it gets deleted from `values`.